core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -718,15 +718,72 @@ private DefaultFunctionResolver time_to_sec() {
 
   /**
    * Extracts the timestamp of a date and time value.
-   * Also to construct a date type. The supported signatures:
-   * STRING/DATE/DATETIME/TIMESTAMP -> DATE
+   * Input strings may contain a timestamp only in format 'yyyy-MM-dd HH:mm:ss[.SSSSSSSSS]'
+   * STRING/DATE/TIME/DATETIME/TIMESTAMP -> TIMESTAMP
+   * STRING/DATE/TIME/DATETIME/TIMESTAMP, STRING/DATE/TIME/DATETIME/TIMESTAMP -> TIMESTAMP
    */
   private DefaultFunctionResolver timestamp() {
     return define(BuiltinFunctionName.TIMESTAMP.getName(),
-        impl(nullMissingHandling(DateTimeFunction::exprTimestamp), TIMESTAMP, STRING),
-        impl(nullMissingHandling(DateTimeFunction::exprTimestamp), TIMESTAMP, DATE),
-        impl(nullMissingHandling(DateTimeFunction::exprTimestamp), TIMESTAMP, DATETIME),
-        impl(nullMissingHandling(DateTimeFunction::exprTimestamp), TIMESTAMP, TIMESTAMP));
+        implWithProperties(nullMissingHandlingWithProperties(DateTimeFunction::exprTimestamp),
+            TIMESTAMP, STRING),
```

Comment:
does this change depends on https://github.com/opensearch-project/sql/pull/1196 ?

Reply:
No, it uses `ADDTIME` function from #1194

Reply:
Sorry I meant with implicit cast changes in https://github.com/opensearch-project/sql/pull/1196, we don't need to permute all date time types, right?

Reply:
Oh, I see. Maybe...
Let me rebase this after #1196 merge and test.

Reply:
I simplified implementation in 51d82cf. Please, have a look, thanks!